view_conversations/conv-viewer-webapi/src/db_types.ts

```diff
@@ -0,0 +1,172 @@
+/*
+ * Copyright 2018 Google Inc.
+ *
+ *Licensed under the Apache License, Version 2.0 (the "License");
+ *you may not use this file except in compliance with the License.
+ *You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *Unless required by applicable law or agreed to in writing, software
+ *distributed under the License is distributed on an "AS IS" BASIS,
+ *WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *See the License for the specific language governing permissions and
+ *limitations under the License.
+*/
+
+import * as spanner from '@google-cloud/spanner';
+import * as ab2str from 'arraybuffer-to-string';
+import * as runtime_types from './runtime_types';
+
+// Spanner Row Representations in JS.
+//
+// Note: Cloud Spanner interprets Node.js numbers as FLOAT64s, so
+// in the row format, they are treated as strings (spanner does
+// the conversion to INT64s when appropriate)
+export interface OutputRow {
+  id: runtime_types.CommentId;
+  ancestor_id: runtime_types.CommentId;
+  authors: string[] | null;
+  cleaned_content: string | null;
+  content: string | null;
+  conversation_id: runtime_types.ConversationId;
+  indentation: number;
+  page_id: number;
+  page_title: string;
+  parent_id: runtime_types.CommentId;
+  replyTo_id: runtime_types.CommentId;
+  rev_id: runtime_types.RevisionId;
+  timestamp: Date | null;
+  type: string;
+  user_id: number | null;
+  user_text: string | null;
+}
+
+
+abstract class SpannerFieldHandler<T> {
+  public fieldName : string;
+  constructor(public newFieldName: string) {this.fieldName = newFieldName}
+  public abstract fromSpannerResultField(row:spanner.ResultField) : T | null;
+}
+
+// TODO(ldixon): add additional constraints, e.g. regexp matching.
+class StringFieldHandler extends SpannerFieldHandler<string> {
+  public fromSpannerResultField(field:spanner.ResultField) : string | null {
+    if(!(field === null || typeof(field) === 'string')) {
+      console.error('Error: field:');
+      console.dir(field);
+      throw Error(`For ${this.fieldName}: expected field:string, not: ${typeof(field)}`);
+    }
+    return field;
+  }
+}
+
+class BytesFieldHandler extends SpannerFieldHandler<string> {
+  public fromSpannerResultField(field:spanner.ResultField) : string | null {
+    if (field === null){
+      return null;
+    }
+    if(!(field instanceof Uint8Array)) {
+        console.error('Error: field:');
+        console.dir(field);
+        throw Error(`For ${this.fieldName}: expected field: byte array, not: ${typeof(field)}`);
+    }
+    return ab2str(field as Uint8Array);
```

Comment:
Do you need this cast? I would think the compiler would assume that field was already a Uint8Array here because of the check above. I noticed this in the other classes as well.

Reply:
Yes, I hit a compile error when doing it without casting.

Reply:
What was the error?

Reply:
Interesting, this particular one works now without casting. But a similar one: parseInt((field as spanner.ResultValue).value, 10)

without casting returns errror: 

src/db_types.ts(103,29): error TS2339: Property 'value' does not exist on type 'string[] | Uint8Array | ResultValue'.